resources/problem-vote.js

```diff
@@ -0,0 +1,91 @@
+var voteChart = null;
+
+function init_vote_form() {
+    $('#delete-vote-form').on('submit', function (e) {
+        e.preventDefault();
+        $.ajax({
+            url: $('#delete-vote-form').prop('action'),
+            type: 'POST',
+            data: $('#delete-vote-form').serialize(),
+            success: function () {
+                $('#vote-button').text(gettext('Vote on problem points'));
+                $.featherlight.close();
+            },
+            error: function (data) {
+                alert(gettext('Unable to delete vote: {error}').replace('{error}', data.responseText));
+            }
+        });
+    });
+
+    $('#vote-form').on('submit', function (e) {
+        e.preventDefault();
+        $.ajax({
+            url: $('#vote-form').prop('action'),
+            type: 'POST',
+            data: $('#vote-form').serialize(),
+            success: function (data) {
+                $('#vote-button').text(gettext('Change vote (%(points)d)').replace('%(points)d', data.points));
+                $.featherlight.close();
+            },
+            error: function (data) {
+                var errors = data.responseJSON;
+                if (errors === undefined) {
+                    alert(gettext('Unable to cast vote: {error}').replace('{error}', data.responseText));
+                }
+
+                $('#points-error').text(typeof errors === 'object' && 'points' in errors ? errors.points[0] : '');
+                $('#note-error').text(typeof errors === 'object' && 'note' in errors ? errors.note[0] : '');
+            }
+        });
+    });
+}
+
+function reload_vote_graph(data, min_possible_vote, max_possible_vote) {
+    if (voteChart !== null) voteChart.destroy();
+
+    // Give the graph some padding on both sides.
+    var min_points = Math.max(data[0] - 2, min_possible_vote);
+    var max_points = Math.min(data[data.length - 1] + 2, max_possible_vote);
+
+    var xlabels = [];
+    var voteFreq = [];
+    for (var i = min_points; i <= max_points; i++) {
+        xlabels.push(i);
+        voteFreq.push(0);
+    }
+
+    data.forEach(function (x) { voteFreq[x - min_points]++; });
```

Comment:
Also, can we just use a dictionary? Maybe that's not super easy... but using a list just feels ugly.

Reply:
dmoj uses chart.js 2.8.0 https://www.chartjs.org/docs/2.8.0/charts/bar.html

chart.js 2.8.0 doesn't support dictionaries.